tests/client/test_v1.py

```diff
@@ -0,0 +1,24 @@
+"""Tests for Shaarli REST API v1 client"""
+from unittest import mock
+
+from shaarli_client.client import ShaarliV1Client
+
+SHAARLI_URL = 'http://domain.tld/shaarli/'
+SHAARLI_SECRET = 's3kr37!'
+
+
+def test_constructor():
+    """Instantiate a new client"""
+    ShaarliV1Client(SHAARLI_URL, SHAARLI_SECRET)
+
+
+@mock.patch('requests.request')
```

Comment:
Right here, this prevents `requests` from actually executing a `GET` query to a non-existent server, by patching the `requests.request` function and replacing it with a dummy callable.

---

Mocks are used to dynamically replace objects with "dummy" ones that can either:
- do nothing
- override the behaviour of the mocked object, e.g. override `requests.get()` with a function that always returns a predefined response

Mocking is a powerful testing practice with many uses:
- avoid being dependent on the "external world", e.g. do not rely on the presence of a web service
- focus on processing rather than on resource access
- focus on testing specific code sections
- check method calls, e.g. how many times something has been called, and with which arguments

Tests usually mock the following operations:
- networking
- file I/O
- standard & 3rd-party library code

This is usually used in conjunction with [pytest fixtures](http://doc.pytest.org/en/latest/fixture.html), with:
- fixtures providing isolated environments: pseudo-filesystem, local servers to test against, etc.
- mock patches overriding undesired operations